examples/apps/generic_client/eth_generic_client.py

```diff
@@ -419,34 +391,53 @@ def _handle_ethereum_event(event, *kargs, **kwargs):
                     .format(decrypted_res))
     sys.exit(1)
 
-
-def _get_worker_details(blockchain_type,
-                        worker_registry, worker_id):
-    # get the worker details for direct/proxy model.
-    # Retrieve worker details
-    jrpc_req_id = 342
-    worker_retrieve_result = worker_registry.worker_retrieve(
-        worker_id, jrpc_req_id
-    )
-    logger.info("\n Worker retrieve response: {}\n".format(
-        json.dumps(worker_retrieve_result, indent=4)
-    ))
-
-    if worker_retrieve_result is None or "error" in worker_retrieve_result:
-        logger.error("Unable to retrieve worker details\n")
-        sys.exit(1)
-
-    # Initializing Worker Object
-    worker_obj1 = worker_details.SGXWorkerDetails()
-    if blockchain_type:
-        worker_obj1.load_worker(
-            json.loads(worker_retrieve_result[4]))
+def _get_first_active_worker(worker_registry, worker_id):
+    """
+    This function looks up all the workers registered. It then filters
+    this data to get the first worker that has Active status.
+    """
+    jrpc_req_id = random.randint(0, 100000)
+    worker_retrieve_result = None
+    if not worker_id:
+        # Lookup all worker id present on  blockchain
+        worker_lookup_result = worker_registry.worker_lookup(
+            WorkerType.TEE_SGX,
+            'aabbcc1234ddeeff',
+            '11aa22bb33cc44dd',
```

Comment:
read values from config file. or you could match any org/app ids

Reply:
@manojgop, raised an [issue](https://github.com/hyperledger/avalon/issues/310) for this. Will remove from elsewhere as well.